test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionTextCacheTest.cs

```diff
@@ -0,0 +1,189 @@
+                    (Expression<Func<TestModel, string>>)(qmodelvar => key)
+                };
+            }
+        }
+
+        public static IEnumerable<object[]> IdenticalExpressions
+        {
+            get
+            {
+                return new TheoryData<Expression, Expression>
+                {
+                    { (Expression<Func<TestModel, Category>>)(model => model.SelectedCategory),
+                        (Expression<Func<TestModel, Category>>)(model => model.SelectedCategory) },
+
+                    { (Expression<Func<TestModel, CategoryName>>)(model => model.SelectedCategory.CategoryName),
+                        (Expression<Func<TestModel, CategoryName>>)(model => model.SelectedCategory.CategoryName) },
+
+                    { (Expression<Func<TestModel, int>>)(testmodel => testmodel.SelectedCategory.CategoryId),
+                        (Expression<Func<TestModel, int>>)(testmodel => testmodel.SelectedCategory.CategoryId) },
+
+                    { (Expression<Func<TestModel, string>>)(model => model.SelectedCategory.CategoryName.MainCategory),
+                        (Expression<Func<TestModel, string>>)(model => model.SelectedCategory.CategoryName.MainCategory) }
+                };
+            }
+        }
+
+        public static IEnumerable<object[]> NonIdenticalExpressions
+        {
+            get
+            {
+                var value = "test";
+                var key = "TestModel";
+
+                return new TheoryData<Expression, Expression>
+                {
+                    { (Expression<Func<TestModel, Category>>)(model => model.SelectedCategory),
+                        (Expression<Func<TestModel, CategoryName>>)(model => model.SelectedCategory.CategoryName) },
+
+                    { (Expression<Func<TestModel, CategoryName>>)(model => model.SelectedCategory.CategoryName),
+                        (Expression<Func<TestModel, string>>)(model => value) },
+
+                    { (Expression<Func<TestModel, string>>)(testmodel => testmodel.SelectedCategory.CategoryName.MainCategory),
+                        (Expression<Func<TestModel, string>>)(testmodel => value) },
+
+                    { (Expression<Func<IList<TestModel>, Category>>)(model => model[2].SelectedCategory),
+                        (Expression<Func<TestModel, string>>)(model => model.SelectedCategory.CategoryName.MainCategory) },
+
+                    { (Expression<Func<TestModel, int>>)(testmodel => testmodel.SelectedCategory.CategoryId),
+                        (Expression<Func<TestModel, Category>>)(model => model.SelectedCategory) },
+
+                    { (Expression<Func<IDictionary<string, TestModel>, string>>)(model => model[key].SelectedCategory.CategoryName.MainCategory),
+                        (Expression<Func<TestModel, Category>>)(model => model.SelectedCategory) }
```

Comment:
Confirm a couple of indexer expressions, even if they look identical, result in `false`.